docs/content/topics/project-developers/secrets.md

```diff
@@ -9,239 +9,157 @@ aliases:
-
-```javascript
-const { events, Job } = require("brigadier");
-
-events.on("exec", () => {
-  var job = new Job("echo", "alpine");
+Once created, the name of the image pull secret may be added to the Worker
+configuration on the intended project. For example, if an image pull secret
+named `privateregistrysecret` exists on the substrate in the namespace for
+project `my-project`, it can be added to the project definition like so:
+
+```yaml
+apiVersion: brigade.sh/v2-beta
+kind: Project
+metadata:
+  id: my-project
+spec:
+  workerTemplate:
+    kubernetes:
+      imagePullSecrets:
+      - privateregistrysecret
+```
 
-  job.env = {
-    mySecretReference: {
-      secretKeyRef: {
-        name: "mysecret",
-        key: "hello"
-      }
-    }
-  };
+Worker and Job containers for events handled by the project will then have
+access to the image pull secret.
 
-  job.tasks = [
-    "echo hello ${mySecretReference}"
-  ];
+[image pull secret]: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+[Operator]: /topics/operators
 
-  job.run();
-});
-```
+## FAQ
 
-To recap, `mySecretReference` is the name of the variable in the job's environment, `secretKeyRef.name` is the name
-of the Kubernetes secret, and `secretKeyRef.key` is the key pointing to the actual secret value (`world` in this example.)
+**Why don't all jobs automatically get access to all of the project's secrets?
+Why do I have to pass them to the `Job.environment`?**
 
-As a result, `mySecretReference` points directly to this secret value in this job's environment.
+Brigade is designed to use off-the-shelf Docker images. In the examples above,
+we used the `debian:latest` image straight from DockerHub. We wouldn't want to
+just automatically pass all of our information straight into that container.
+For starters, doing so might inadvertently override an existing environment
+variable of the same name. More importantly, the data might get misused or
+unintentionally exposed by the container.
 
-To run this example, we replace `BRIGADE_PROJECT_NAME` with our Brigade project name and run:
+So we err on the side of safety.
 
-```console
-brig run BRIGADE_PROJECT_NAME -f brigade.js
-```
+**Can I encrypt my secrets?**
 
-We will then see `"hello world"` displayed in the logs.
+Brigade uses [Kubernetes Secrets] for holding sensitive data. Kubernetes Secret
+data is base64-encoded but not encrypted by default. Operators looking to
+encrypt these resources should reference the [Kubernetes documentation]. Note
+that support for this feature will depend on the distribution used (e.g. the
+cloud provider offering or other self-hosted option.)
+
+Another option for Brigade developers to access secret values is via a trusted
+key store external to Kubernetes, such as HashiCorp's [Vault] or [Azure Key
+Vault]. The scripting logic can then use the appropriate CLI or API calls to
+fetch secrets in the event handler(s) that require them.
```

Comment:
This is an interesting option, but... any secrets you use will _still_ end up on the substrate while the jobs run. And beyond that, you probably need to use Brigade secrets to store key vault creds...

So at the end of the day, even this is no more secure than whatever your k8s distro offers.

Reply:
True.  Removed this section for now.